File: examples/fr_stats.py
# This file contains statistics functions written with French comments
# Un script simple pour calculer des statistiques de base sur une liste de nombres

# Définir une liste de nombres exemples
numbers = [4, 8, 15, 16, 23, 42]

# Calculer la somme totale de tous les nombres
total = sum(numbers)

average = total / len(numbers)  # Calculer la moyenne en divisant la somme par le nombre d'éléments

minimum = min(numbers)  # Trouver la valeur minimale dans la liste
maximum = max(numbers)  # Trouver la valeur maximale dans la liste

# Afficher les résultats dans la console
print(f"Numbers: {numbers}")
print(f"Sum:     {total}")
print(f"Average: {average:.2f}")
print(f"Min:     {minimum}")
print(f"Max:     {maximum}")

File: examples/fr_stats.java
// This file contains statistics functions written with French comments

/**
 * Calcule les statistiques de base d'une liste de nombres.
 * Affiche la somme, la moyenne, le minimum et le maximum.
 */
public class fr_stats {

    /** Calcule la somme de tous les éléments du tableau */
    static int sum(int[] arr) {
        int total = 0; // Valeur initiale de la somme
        for (int n : arr) {
            total += n; // Ajouter chaque élément à la somme
        }
        return total;
    }

    /** Trouve la valeur minimale dans le tableau */
    static int minimum(int[] arr) {
        int min = arr[0];
        for (int n : arr) {
            if (n < min) { min = n; } // Mettre à jour le minimum si nécessaire
        }
        return min;
    }

    /** Trouve la valeur maximale dans le tableau */
    static int maximum(int[] arr) {
        int max = arr[0];
        for (int n : arr) {
            if (n > max) { max = n; } // Mettre à jour le maximum si nécessaire
        }
        return max;
    }

    public static void main(String[] args) {
        /* ------ Définition des données ------ */
        int[] numbers = {4, 8, 15, 16, 23, 42}; // Tableau de nombres exemples

        int total = sum(numbers);
        double average = (double) total / numbers.length; // Diviser la somme par le nombre d'éléments

        double min = minimum(numbers); // Valeur minimale de la liste
        double max = maximum(numbers); // Valeur maximale de la liste

        /* ------ Affichage des résultats ------ */
        /*
         * Les résultats sont formatés pour la lisibilité.
         */
        System.out.println("Nombres: [4, 8, 15, 16, 23, 42]");
        System.out.printf("Somme:   %d%n",   total);
        System.out.printf("Moyenne: %.2f%n", average);
        System.out.printf("Valeur minimale: %.0f%n", min);
        System.out.printf("Valeur maximale: %.0f%n", max);
    }
}
